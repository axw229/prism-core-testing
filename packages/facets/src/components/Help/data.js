// @flow
import { shouldAllowFeature } from '../../shared/utils/featureSwitch.util'
import { FEATURE_EXCLUSIONS } from '../../constants/configurations'

export const helpTabs = [{
  'id': 0,
  'name': 'ICONS_&_BUTTONS',
  'header': 'HELPFUL_HINTS.HEADER.ICONS_&_BUTTONS',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.ICONS_&_BUTTONS',
  'content': [{
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'plus-circle', 'rotate': 0 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.ADD_COLOR',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.ADD_COLOR']
  }, {
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'folder', 'rotate': 0 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.SAVE',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.SAVE']
  }, {
    'fontAwesomeIcon': { 'variant': 'fa', 'icon': 'trash', 'rotate': 0 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.TRASH_A_COLOR',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.TRASH_A_COLOR']
  }, {
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'clone', 'rotate': 0, 'flip': 'horizontal' },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.MORE_SCENES',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.MORE_SCENES']
  }, {
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'align-justify', 'rotate': 0, 'flip': 'horizontal' },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.COLOR_DETAILS',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.COLOR_DETAILS']
  }, {
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'signal-3', 'rotate': 235, 'size': 'lg' },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.GRAB_&_REORDER',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.GRAB_&_REORDER']
  }, {
    'fontAwesomeIcon': {},
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.PAINT_SCENE',
    'iconInfoContent': []
  }],
  'imageList': '',
  'isHiddenMobile': false
}, {
  'id': 1,
  'name': 'MY_COLOR_PALETTE',
  'header': 'HELPFUL_HINTS.HEADER.MY_COLOR_PALETTE',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.MY_COLOR_PALETTE',
  'content': '',
  'imageList': [{
    'alt': '',
    'role': '',
    'imagePathKey': 'myColorPaletteBg'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.CLICK_THE_COLOR_DETAILS_ICON_TO',
    'role': 'text',
    'imagePathKey': 'myColorPalette1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.CLICK_THE_TRASH_ICON_TO',
    'role': 'text',
    'imagePathKey': 'myColorPalette2'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.AN_ACTIVE_COLOR_STATE',
    'role': 'text',
    'imagePathKey': 'myColorPalette3'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.TAP_AND_HOLD_THE_BOTTOM_EDGE_OF_A_COLOR_SWATCH_TO',
    'role': 'text',
    'imagePathKey': 'myColorPalette4'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.CLICK_THE_ADD_COLOR_ICON_TO',
    'role': 'text',
    'imagePathKey': 'myColorPalette5'
  }],
  'subContent': [{
    'header': 'HELPFUL_HINTS.SUB_CONTENT_HEADER.MY_COLOR_PALETTE',
    'content': 'HELPFUL_HINTS.SUB_CONTENT_DESCRIPTION.MY_COLOR_PALETTE'
  }],
  'imageListMobile': [{
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.CLICK_THE_COLOR_DETAILS_ICON_TRASH_ICON_TO',
    'role': 'text',
    'imagePathKey': 'myColorPaletteMobile1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.AN_ACTIVE_COLOR_STATE',
    'role': 'text',
    'imagePathKey': 'myColorPaletteMobile2'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.CLICK_THE_ADD_COLOR_ICON_TO',
    'role': 'text',
    'imagePathKey': 'myColorPaletteMobile3'
  }],
  'isHiddenMobile': false
}, {
  'id': 2,
  'name': 'ADDING_COLORS',
  'header': 'HELPFUL_HINTS.HEADER.ADDING_COLORS',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.ADDING_COLORS',
  'content': '',
  'imageList': [{
    'alt': '',
    'role': '',
    'imagePathKey': 'addColorBg'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.CLICK_ADD_A_COLOR_ICON_IN_A_COLOR_TILE_TO',
    'role': 'text',
    'imagePathKey': 'addColor1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.CLICK_THE_ADD_COLOR_ICON_TO',
    'role': 'text',
    'imagePathKey': 'addColor2'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.START_A_NEW_PALETTE_TO_ADD_MORE_THAN_EIGHT_COLORS',
    'role': 'text',
    'imagePathKey': 'addColor3'
  }],
  'subContent': [{
    'header': 'HELPFUL_HINTS.SUB_CONTENT_HEADER.ADDING_COLORS',
    'content': 'HELPFUL_HINTS.SUB_CONTENT_DESCRIPTION.ADDING_COLORS'
  }],
  'imageListMobile': [{
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.CLICK_ADD_A_COLOR_ICON_IN_A_COLOR_TILE_TO',
    'role': 'text',
    'imagePathKey': 'addColorMobile1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.CLICK_THE_ADD_COLOR_ICON_TO',
    'role': 'text',
    'imagePathKey': 'addColorMobile2'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.START_A_NEW_PALETTE_TO_ADD_MORE_THAN_EIGHT_COLORS',
    'role': 'text',
    'imagePathKey': 'addColorMobile3'
  }],
  'isHiddenMobile': false
}, {
  'id': 3,
  'name': 'COLOR_DETAILS',
  'header': 'HELPFUL_HINTS.HEADER.COLOR_DETAILS',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.COLOR_DETAILS',
  'content': '',
  'imageList': [{
    'alt': '',
    'role': '',
    'imagePathKey': 'colorDetailBg'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.COLOR_DETAILS.CLICK_THE_COLOR_DETAILS_ICON_TO',
    'role': 'text',
    'imagePathKey': 'colorDetail1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.COLOR_DETAILS.PREVIEW_A_COLOR_IN_COLOR_DETAILS',
    'role': 'text',
    'imagePathKey': 'colorDetail2'
  }],
  'imageListMobile': [{
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.COLOR_DETAILS.CLICK_THE_COLOR_DETAILS_ICON_TO',
    'role': 'text',
    'imagePathKey': 'colorDetailMobile1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.COLOR_DETAILS.PREVIEW_A_COLOR_IN_COLOR_DETAILS',
    'role': 'text',
    'imagePathKey': 'colorDetailMobile2'
  }],
  'isHiddenMobile': false
}, {
  'id': 4,
  'name': 'PAINTING_MY_OWN_PHOTO',
  'header': 'HELPFUL_HINTS.HEADER.PAINTING_MY_OWN_PHOTO',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.PAINTING_MY_OWN_PHOTO',
  'content': [{
    'fontAwesomeIcon': { 'variant': 'fa', 'icon': 'fill-drip', 'rotate': 0, 'flip': 'horizontal' },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.PAINT_AREA',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.PAINT_AREA']
  }, {
    'fontAwesomeIcon': { 'variant': 'fa', 'icon': 'brush', 'rotate': 45 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.PAINTBRUSH',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.PAINTBRUSH']
  }, {
    'fontAwesomeIcon': { 'variant': 'fa', 'icon': 'mouse-pointer', 'rotate': -20, 'flip': 'horizontal' },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.SELECT',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.SELECT']
  }, {
    'fontAwesomeIcon': { 'variant': 'fa', 'icon': 'eraser', 'rotate': 0 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.ERASE',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.ERASE']
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'draw-polygon', 'rotate': 10 }, {
      'variant': 'fal',
      'icon': 'plus',
      'rotate': 0,
      'size': 'xs'
    }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.DEFINE_AREA',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.DEFINE_AREA']
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'draw-polygon', 'rotate': 10 }, {
      'variant': 'fal',
      'icon': 'minus',
      'rotate': 0,
      'size': 'xs'
    }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.REMOVE_AREA',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.REMOVE_AREA']
  }, {
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'search-plus', 'rotate': 0 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.ZOOM',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.ZOOM']
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fa', 'icon': 'undo-alt', 'rotate': 0 }, {
      'variant': 'fa',
      'icon': 'redo-alt',
      'rotate': 0
    }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.UNDO_&_REDO',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.UNDO_&_REDO'],
    'isUndoRedo': true
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fa', 'icon': 'redo-alt' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.REDO',
    'iconInfoContent': []
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fa', 'icon': 'eye', 'flip': 'horizontal' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.HIDE_PAINT',
    'iconInfoContent': []
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'info-circle' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.INFO',
    'iconInfoContent': []
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'trash-alt' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.DELETE_GROUP',
    'iconInfoContent': []
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'object-group' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.GROUP',
    'iconInfoContent': []
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'object-ungroup' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.UNGROUP',
    'iconInfoContent': []
  }],
  'imageList': '',
  'isHiddenMobile': false
}, {
  'id': 5,
  'name': 'SAVING_MY_WORK',
  'header': 'HELPFUL_HINTS.HEADER.SAVING_MY_WORK',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.SAVING_MY_WORK',
  'content': '',
  'imageList': [{
    'alt': '',
    'role': '',
    'imagePathKey': 'savingBg'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.SAVING_MY_WORK.NAME_YOUR_CURRENT_PHOTO',
    'role': 'text',
    'imagePathKey': 'saving1'
  }, { 'alt': 'HELPFUL_HINTS.IMAGE_ALT.SAVING_MY_WORK.CLICK_SAVE_ICON_TO', 'role': 'text', 'imagePathKey': 'saving3' }],
  'imageListMobile': [{
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.SAVING_MY_WORK.CLICK_SAVE_ICON_TO',
    'role': 'text',
    'imagePathKey': 'savingMobile1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.SAVING_MY_WORK.NAME_YOUR_CURRENT_PHOTO',
    'role': 'text',
    'imagePathKey': 'savingMobile2'
  }],
  'isHiddenMobile': false
}]

export const helpHeader = 'HELPFUL_HINTS.TITLE'

export const filterHelpItems = (contents: string[], contentsHiddenMobile: string[], featureExclusions: string[]): typeof helpTabs => {
  // Filtering Help Section details:
  // Icons & Buttons - Save hint
  !shouldAllowFeature(featureExclusions, FEATURE_EXCLUSIONS.documentSaving) && helpTabs[0].content.forEach((icon, index) => {
    icon.iconInfoName === 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.SAVE' && helpTabs[0].content.splice(index, 1)
  })

  const helpTabNames = helpTabs.map(item => item.name)
  contentsHiddenMobile.forEach(content => { helpTabs[helpTabNames.indexOf(content)].isHiddenMobile = true })

  return helpTabs.filter(item => contents.indexOf(item.name) !== -1)
}
