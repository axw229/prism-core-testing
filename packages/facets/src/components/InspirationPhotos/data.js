// @flow
import tinycolor from '@ctrl/tinycolor'
import { type Color } from 'src/shared/types/Colors.js.flow'
import uniqueId from 'lodash/uniqueId'

export const throttleDragTime = 5
// we need this two const to make sure cursor always point to the center of preview circle
export const activedPinsHalfWidth = 24

export const cloneColorPinsArr = (pins) => {
  return pins.map(pin => ({ ...pin }))
}

export const getRGBInitPins = (acgColors: Array<Object>) => {
  return acgColors.map<Object>((acgColor: any): Object => {
    return {
      rgbArray: [acgColor.r, acgColor.g, acgColor.b],
      rgbValue: `rgb(${acgColor.r},${acgColor.g},${acgColor.b})`,
      translateValueX: acgColor.x,
      translateValueY: acgColor.y
    }
  })
}

export const rgb2lab = (rgb: Array<any>) => {
  let r = rgb[0] / 255
  let g = rgb[1] / 255
  let b = rgb[2] / 255
  let x; let y; let z

  r = (r > 0.04045) ? Math.pow((r + 0.055) / 1.055, 2.4) : r / 12.92
  g = (g > 0.04045) ? Math.pow((g + 0.055) / 1.055, 2.4) : g / 12.92
  b = (b > 0.04045) ? Math.pow((b + 0.055) / 1.055, 2.4) : b / 12.92

  x = (r * 0.4124 + g * 0.3576 + b * 0.1805) / 0.95047
  y = (r * 0.2126 + g * 0.7152 + b * 0.0722) / 1.00000
  z = (r * 0.0193 + g * 0.1192 + b * 0.9505) / 1.08883

  x = (x > 0.008856) ? Math.pow(x, 1 / 3) : (7.787 * x) + 16 / 116
  y = (y > 0.008856) ? Math.pow(y, 1 / 3) : (7.787 * y) + 16 / 116
  z = (z > 0.008856) ? Math.pow(z, 1 / 3) : (7.787 * z) + 16 / 116

  return [(116 * y) - 16, 500 * (x - y), 200 * (y - z)]
}

export type RenderingPin = {
  isActiveFlag: boolean,
  name: string,
  colorNumber: number,
  rgbValue: string,
  pinNumber: number,
  isContentLeft: boolean,
  translateX: number,
  translateValueY: number,
  pinId: string
}

export const renderingPins = (initPins: Array<Object>, canvasOffsetWidth: number, canvasOffsetHeight: number, colors: Color[]): RenderingPin[] => {
  return getRGBInitPins(initPins).map<Object>((acgColor: any, index: number): Object => {
    const color: Color = findClosestColor(acgColor.rgbArray, colors)
    const calculateTranslateX = acgColor.translateValueX * canvasOffsetWidth
    const calculateTranslateY = acgColor.translateValueY * canvasOffsetHeight

    return {
      ...color,
      isActiveFlag: (initPins.length - 1 === index),
      rgbValue: `rgb(${color.red},${color.green},${color.blue})`,
      pinNumber: index,
      pinId: uniqueId('pin_'),
      isContentLeft: calculateTranslateX < canvasOffsetWidth / 2,
      translateX: Math.min(canvasOffsetWidth - 2 * activedPinsHalfWidth, calculateTranslateX),
      translateY: Math.min(canvasOffsetHeight - 2 * activedPinsHalfWidth, calculateTranslateY)
    }
  })
}

export const findClosestColor = ([red, green, blue]: [number, number, number], colors: Color[]): Color => {
  const currentPixelHex: string = tinycolor(`rgb (${red}, ${green}, ${blue})`).toHex()

  return colors.reduce(([closestColor, smallestDistanceSoFar], color): [Color, number] => {
    const colorDistance: number = measureDistance(currentPixelHex, tinycolor(`rgb(${color.red},${color.green},${color.blue})`).toHex())
    return colorDistance < smallestDistanceSoFar ? [color, colorDistance] : [closestColor, smallestDistanceSoFar]
  }, [colors[0], 99999])[0]
}

export const measureDistance = (color1: string, color2: string): number => {
  const position1 = getChromaLuma(color1)
  const position2 = getChromaLuma(color2)
  const redDelta = position1[0] - position2[0]
  const greenDelta = position1[1] - position2[1]
  const lumaDelta = position1[2] - position2[2]
  const distance = redDelta * redDelta + greenDelta * greenDelta + lumaDelta * lumaDelta
  return distance
}

export const getChromaLuma = (rgbString: string) => {
  const color = tinycolor(rgbString)
  const luminosity = color.getLuminance()
  const colorRgb = color.toRgb()
  const red = colorRgb.r
  const green = colorRgb.g
  const blue = colorRgb.b
  const sum = red + green + blue
  let redChroma
  let greenChroma
  if (sum > 25) {
    redChroma = red / sum
    greenChroma = green / sum
  } else {
    redChroma = 1 / 3
    greenChroma = 1 / 3
  }

  return [redChroma, greenChroma, luminosity]
}
