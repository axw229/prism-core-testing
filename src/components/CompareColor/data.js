export const Scene = {
  'id': 1,
  'width': 1311,
  'height': 792,
  'variant_names': [
    'day',
    'night'
  ],
  'variants': [
    {
      'name': 'Living Room Day',
      'variant_name': 'day',
      'image': 'https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom6_day?wid=1311}&qlt=92',
      'thumb': 'https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom6_day?wid=300}&qlt=80',
      'normalizedImageValueCurve': '0 0.1 0.15 0.25 0.4 0.5 0.6 0.7 0.84 0.955 1',
      'surfaces': [
        {
          'id': 1,
          'mask': 'https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom6_day?wid=1280&req=object&opac=100&fmt=png8&object=wall&color=000000}&fmt=png8&bgColor=FFFFFF&op_invert=1',
          'hitArea': '/prism/images/scenes/rooms/2/m1.svg',
          'role': 'main'
        },
        {
          'id': 2,
          'mask': 'https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom6_day?wid=1280&req=object&opac=100&fmt=png8&object=accent&color=000000}&fmt=png8&bgColor=FFFFFF&op_invert=1',
          'hitArea': '/prism/images/scenes/rooms/2/m2.svg',
          'role': 'main'
        },
        {
          'id': 3,
          'mask': 'https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom6_day?wid=1280&req=object&opac=100&fmt=png8&object=trim&color=000000}&fmt=png8&bgColor=FFFFFF&op_invert=1',
          'hitArea': '/prism/images/scenes/rooms/2/m3.svg',
          'role': 'trim'
        }
      ]
    },
    {
      'name': 'Living Room Night',
      'variant_name': 'night',
      'image': 'https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom6_night?wid=1311}&qlt=92',
      'thumb': 'https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom6_night?wid=300}&qlt=80',
      'normalizedImageValueCurve': '0 0.1 0.22 0.32 0.42 0.51 0.6 0.7 0.84 0.955 1',
      'surfaces': [
        {
          'id': 1,
          'mask': 'https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom6_night?wid=1280&req=object&opac=100&fmt=png8&object=wall&color=000000}&fmt=png8&bgColor=FFFFFF&op_invert=1',
          'hitArea': '/prism/images/scenes/rooms/2/m1.svg',
          'role': 'main'
        },
        {
          'id': 2,
          'mask': 'https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom6_night?wid=1280&req=object&opac=100&fmt=png8&object=accent&color=000000}&fmt=png8&bgColor=FFFFFF&op_invert=1',
          'hitArea': '/prism/images/scenes/rooms/2/m2.svg',
          'role': 'main'
        },
        {
          'id': 3,
          'mask': 'https://sherwin.scene7.com/is/image/sw?src=ir{swRender/hd_livingroom6_night?wid=1280&req=object&opac=100&fmt=png8&object=trim&color=000000}&fmt=png8&bgColor=FFFFFF&op_invert=1',
          'hitArea': '/prism/images/scenes/rooms/2/m3.svg',
          'role': 'trim'
        }
      ]
    }
  ]
}
