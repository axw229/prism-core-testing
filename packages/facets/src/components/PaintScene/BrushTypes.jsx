// @flow
import React from 'react'
import { useIntl } from 'react-intl'
import { brushLargeSize, brushMediumSize, brushRoundShape, brushShapesTypes,brushSmallSize, brushTinySize } from './data'
import RadioButton from './RadioButton'
import './BrushTypes.scss'

const baseClass = 'brush-types'
const wrapperClass = `${baseClass}__wrapper`
const shapesContainerClass = `${baseClass}__shapes-container`
const brushButtonClass = `${baseClass}__brush-button`
const brushButtonCircleClass = `${brushButtonClass}--circle`
const brushButtonLargeClass = `${brushButtonClass}--large`
const brushButtonMediumClass = `${brushButtonClass}--medium`
const brushButtonSmallClass = `${brushButtonClass}--small`
const brushButtonTinyClass = `${brushButtonClass}--tiny`
const brushButtonActiveClass = `${brushButtonClass}--active`
const ARIA_LABEL = 'PAINT_TOOLS.ICON_SELECTED_INFO'

type Props = {
  activeWidth: number,
  activeShape: string,
  setBrushShapeSize: Function,
  brushTypeName: string
}

export function BrushTypes ({ activeWidth, activeShape, setBrushShapeSize, brushTypeName }: Props) {
  const intl = useIntl()
  const brushTypesRefs = brushShapesTypes.reduce((acc, value) => {
    acc[value.id] = React.createRef()
    return acc
  }, {})

  const enableBrushType = (isNext: boolean, ref: Object) => {
    if (isNext) {
      if (ref.current === brushTypesRefs[7].current) {
        brushTypesRefs[0].current.focus()
        brushTypesRefs[0].current.click()
      } else {
        brushShapesTypes.forEach((brushElement, index) => {
          if (ref.current === brushTypesRefs[index].current) {
            brushTypesRefs[index + 1].current.click()
            brushTypesRefs[index + 1].current.focus()
          }
        })
      }
    } else if (!isNext) {
      if (ref.current === brushTypesRefs[0].current) {
        brushTypesRefs[7].current.focus()
        brushTypesRefs[7].current.click()
      } else {
        brushShapesTypes.forEach((brushElement, index) => {
          if (ref.current === brushTypesRefs[index].current) {
            brushTypesRefs[index - 1].current.focus()
            brushTypesRefs[index - 1].current.click()
          }
        })
      }
    }
  }

  const getBrushTypes = (activeWidth: number, activeShape: string) => {
    return (
      brushShapesTypes.map((brushShapeType: Object, index: number) => {
        const brushType = brushShapeType.size
        const brushShape = brushShapeType.shape
        let brushClass = `${brushButtonClass}`

        if (brushShape === brushRoundShape) {
          brushClass += ` ${brushButtonCircleClass}`
        }

        if (brushType === brushLargeSize) {
          brushClass += ` ${brushButtonLargeClass}`
        } else if (brushType === brushMediumSize) {
          brushClass += ` ${brushButtonMediumClass}`
        } else if (brushType === brushSmallSize) {
          brushClass += ` ${brushButtonSmallClass}`
        } else if (brushType === brushTinySize) {
          brushClass += ` ${brushButtonTinyClass}`
        }

        if ((activeWidth === brushType && activeShape === brushShape)) {
          brushClass += ` ${brushButtonActiveClass}`
        }

        const triggerSetBrushShapeSize = () => {
          setBrushShapeSize(brushShape, brushType)
        }

        return <RadioButton
          className={`${brushClass}`}
          key={`${brushShape}-${brushType}`}
          triggerSetBrushShapeSize={triggerSetBrushShapeSize}
          enableBrushType={enableBrushType}
          isSelected={(activeWidth === brushType && activeShape === brushShape)}
          value={`${brushShape}-${brushType}`}
          ariaLabel={intl.formatMessage({ id: ARIA_LABEL }, { brushType: brushType, brushShape: brushShape, brushTypeName: brushTypeName })}
          id={`${brushTypeName}${index}`}
          ref={brushTypesRefs[index]}
          brushTypeName={brushTypeName}
        />
      })
    )
  }

  return (
    <div className={`${wrapperClass}`} role='radiogroup' aria-labelledby={`${brushTypeName}BrushType`}>
      <div className={'visually-hidden'} id={`${brushTypeName}BrushType`}>{`${brushTypeName} Brush Types`}</div>
      <div className={`${shapesContainerClass}`}>
        {getBrushTypes(activeWidth, activeShape)}
      </div>
    </div>
  )
}

export default BrushTypes
