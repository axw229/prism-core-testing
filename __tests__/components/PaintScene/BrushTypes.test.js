import React from 'react'
import BrushTypes from 'src/components/PaintScene/BrushTypes'
import { brushLargeSize, brushMediumSize, brushSmallSize, brushTinySize, brushRoundShape, brushSquareShape } from 'src/components/PaintScene/data'
import { fireEvent } from '@testing-library/dom'

it('BrushTypes', async () => {
  const brushSizeArray = [brushLargeSize, brushMediumSize, brushSmallSize, brushTinySize]
  const brushShapeTypeArray = [brushRoundShape, brushSquareShape]
  const setBrushShapeSize = jest.fn()
  const { getByLabelText } = render(
    <BrushTypes
      activeWidth={brushSizeArray[0]}
      activeShape={brushShapeTypeArray[0]}
      setBrushShapeSize={setBrushShapeSize}
    />
  )

  brushSizeArray.map(brushSize => {
    if (brushSize === brushSizeArray[0]) {
      getByLabelText(`the ${brushSize} pixel wide, ${brushShapeTypeArray[0]} brush is active`)
    } else {
      getByLabelText(`select the ${brushSize} pixel wide, ${brushShapeTypeArray[0]} brush`)
    }
  })

  brushSizeArray.map(brushSize => {
    getByLabelText(`select the ${brushSize} pixel wide, ${brushShapeTypeArray[1]} brush`)
  })

  await fireEvent.click(getByLabelText(`select the ${brushMediumSize} pixel wide, ${brushShapeTypeArray[1]} brush`))
  expect(setBrushShapeSize).toHaveBeenCalled()
})
