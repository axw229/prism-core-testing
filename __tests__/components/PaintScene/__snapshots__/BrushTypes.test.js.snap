// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrushTypes render & event testing should match snapshot 1`] = `
<div
  className="brush-types__wrapper"
>
  <div
    className="brush-types__circle-shapes-container"
  >
    <button
      className="brush-types__brush-button brush-types__brush-button--circle brush-types__brush-button--large brush-types__brush-button--active"
      key="round-38"
      onClick={[Function]}
    />
    <button
      className="brush-types__brush-button brush-types__brush-button--circle brush-types__brush-button--medium"
      key="round-30"
      onClick={[Function]}
    />
    <button
      className="brush-types__brush-button brush-types__brush-button--circle brush-types__brush-button--small"
      key="round-22"
      onClick={[Function]}
    />
    <button
      className="brush-types__brush-button brush-types__brush-button--circle brush-types__brush-button--tiny"
      key="round-14"
      onClick={[Function]}
    />
  </div>
  <div
    className="brush-types__square-shapes-container"
  >
    <button
      className="brush-types__brush-button brush-types__brush-button--large"
      key="square-38"
      onClick={[Function]}
    />
    <button
      className="brush-types__brush-button brush-types__brush-button--medium"
      key="square-30"
      onClick={[Function]}
    />
    <button
      className="brush-types__brush-button brush-types__brush-button--small"
      key="square-22"
      onClick={[Function]}
    />
    <button
      className="brush-types__brush-button brush-types__brush-button--tiny"
      key="square-14"
      onClick={[Function]}
    />
  </div>
</div>
`;
